docs/API.md

```diff
@@ -315,6 +317,23 @@ $> gulp --tasks
 
 #### Async support
 
+##### ES2017 async function
+
+ES2017 standardized [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function). Such function returns by definition always a promise. The javascript engine wraps an async function into a promise and enures that such a promise will be automatically resolved when the async function returns or it will be rejected upon any uncaught exception. 
+
+This makes the explicit done() call required by gulp 4 obsolete and avoids the "Did you forget to signal async completion?" error message. Further more it allows you to write asynchonous code in a synchonous style by using the await statement. 
```

Comment:
Thanks for the PR, it's a good idea to mention this.

Gulp (both 3 and 4) support tasks returning Promises. Using async function is more a replacement to returning a Promise than using a `done` callback. I'd just reword this paragraph to say that it provides a lighter syntax over Promises.

Reply:
Yeah, we handle all sorts of async completion (streams, child process, promises, event emitters, observables & callbacks) so I don't want to make it sound like you can't just return a promise.